### PR TITLE
feat: Impl get user info & Add test code

### PR DESCRIPTION
**Pull request by naekang**
### 사용자 정보 조회 및 테스트 코드 작성
**AS-IS**
- SolvedAc 닉네임으로 사용자 정보 조회 기능 구현
- 테스트 코드 작성
- 회원 가입 시 SolvedAc 닉네임 중복 확인 및 예외 처리

**TO-BE**
- 사용자 프로필 이미지 등록 기능 구현
- S3 연결

### 테스트 
- [x] 테스트 코드
- [ ] API 테스트

### CODE REVIEW

**Comment by naekang** [comment]
|File|Coverage [55.85%]|:x:|
|:-|:-:|:-:|
|[ErrorCode.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fexception%2FErrorCode.java)|86.32%|:green_apple:|
|[GetUserInfosResDTO.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fdto%2FGetUserInfosResDTO.java)|77.92%|:x:|
|[User.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fentity%2FUser.java)|60.9%|:x:|
|[AuthService.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fservice%2FAuthService.java)|53.19%|:x:|
|[UserController.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fcontroller%2FUserController.java)|50%|:x:|
|[UserService.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fservice%2FUserService.java)|39.29%|:x:|
|[BaekjoonUserScraper.java](https://github.com/naekang/wealgo-project/blob/4b8b5ad0997181b9e96e0a052828ac26845d1d92/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Futil%2Fscraper%2FBaekjoonUserScraper.java)|5%|:x:|

|Total Project Coverage|39.6%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/naekang/wealgo/domain/user/service/UserService.java [hunk 1]
```diff
@@ -25,8 +32,30 @@ public class UserService {
 
     private static final String apiUrl = "https://solved.ac/api/v3/user/problem_stats?handle=naekang";
 
+    private final AuthRepository authRepository;
+
     private final UserProblemStatsRepository userProblemStatsRepository;
 
+    @Transactional(readOnly = true)
+    public GetUserInfosResDTO getUserInfosByUsername(String username) {
```
Comment (KimHyoJin) [inline src/main/java/com/naekang/wealgo/domain/user/service/UserService.java:40]: view에서 사용하는 객체가 service layer까지 내려오지 않는것도 한번 고민해보시면 좋을 것 같아요. 서비스가 서비스를 참조하는 경우가 생길수도 있기때문에 일반적으로는 서비스에서 사용하는 클래스와 controller에서 response로 내려주는 클래스를 분리하여 계층을 좀더 명확히 하는편입니다.

File: src/main/java/com/naekang/wealgo/domain/auth/entity/User.java [hunk 2]
```diff
@@ -37,14 +37,15 @@ public class User extends BaseEntity {
...
+    @Column(unique = true, name = "SOLVED_AC_USERNAME")
     private String username;
 
     private String password;
 
     @Enumerated(EnumType.STRING)
     private UserRole role;
 
-    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```
Comment (KimHyoJin) [inline src/main/java/com/naekang/wealgo/domain/auth/entity/User.java:48]: 요 정보를 eager로 fetch하게 된 이유가 있을까요?

**Review by KimHyoJin — APPROVED** [review]
안녕하세요 밍밍멘토입니다. 전체적으로 깔끔하게 잘 짜시는것 같아요! 한가지 추가적인 리뷰 드렸습니다. 수고 많으셨어요

File: src/main/java/com/naekang/wealgo/domain/user/service/UserService.java (hunk 1 above)
Comment (naekang) [reply src/main/java/com/naekang/wealgo/domain/user/service/UserService.java:40]: @KimHyoJin 리뷰 감사합니다!
그렇다면 현재 `String` 타입의 파라미터인 `username`을 새로운 RequestDTO로 만든 후 Service layer에서 사용하게 된다면 **service 사용하는 클래스**와 **controller response 클래스**의 분리를 잘 이루었다고 생각하면 될까요?

File: src/main/java/com/naekang/wealgo/domain/auth/entity/User.java (hunk 2 above)
Comment (naekang) [reply src/main/java/com/naekang/wealgo/domain/auth/entity/User.java:48]: 현재 OneToOne 단방향 매핑이 되어있는데 양방향과 착각하여 N+1 문제가 발생할 수 있다고 판단해 `EAGER` 타입으로 변경하였습니다.

수정하여 다시 올리겠습니다!

**Comment by naekang** [comment]
|File|Coverage [55.85%]|:x:|
|:-|:-:|:-:|
|[ErrorCode.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fexception%2FErrorCode.java)|86.32%|:green_apple:|
|[GetUserInfosResDTO.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fdto%2FGetUserInfosResDTO.java)|77.92%|:x:|
|[User.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fentity%2FUser.java)|60.9%|:x:|
|[AuthService.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fservice%2FAuthService.java)|53.19%|:x:|
|[UserController.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fcontroller%2FUserController.java)|50%|:x:|
|[UserService.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fservice%2FUserService.java)|39.29%|:x:|
|[BaekjoonUserScraper.java](https://github.com/naekang/wealgo-project/blob/579a1626ba7b90bea467b60ae2bf93e51bc87631/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Futil%2Fscraper%2FBaekjoonUserScraper.java)|5%|:x:|

|Total Project Coverage|39.6%|:x:|
|:-|:-:|:-:|

**Comment by naekang** [comment]
|File|Coverage [43.68%]|:x:|
|:-|:-:|:-:|
|[ErrorCode.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fexception%2FErrorCode.java)|86.32%|:green_apple:|
|[User.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fentity%2FUser.java)|58.65%|:x:|
|[AuthService.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fauth%2Fservice%2FAuthService.java)|53.19%|:x:|
|[UserController.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fcontroller%2FUserController.java)|40%|:x:|
|[UserService.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fservice%2FUserService.java)|26.09%|:x:|
|[BaekjoonUserScraper.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Futil%2Fscraper%2FBaekjoonUserScraper.java)|5%|:x:|
|[GetUserInfosResponseDTO.java](https://github.com/naekang/wealgo-project/blob/9673cb8ac38187d35b4f169816964bb1a1c4fa8c/src%2Fmain%2Fjava%2Fcom%2Fnaekang%2Fwealgo%2Fdomain%2Fuser%2Fcontroller%2Fresponse%2FGetUserInfosResponseDTO.java)|0%|:x:|

|Total Project Coverage|36.74%|:x:|
|:-|:-:|:-:|

File: src/main/java/com/naekang/wealgo/domain/user/service/UserService.java (hunk 1 above)
Comment (KimHyoJin) [reply src/main/java/com/naekang/wealgo/domain/user/service/UserService.java]: 네네 응답시점에 조립하여 반환하는것이 분리가 된 구조라고 말할수 있겠습니다

Comment (naekang) [reply src/main/java/com/naekang/wealgo/domain/user/service/UserService.java]: 넵 감사합니다! :)

**Pull request merged**